lib/assets/SnapshotAlertManager.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  SnapshotAlertManager.swift
+//
+//  Created by Maxime Britto on 06/01/2016.
+//  Copyright © 2016 dev2a. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+func automaticallyDismissAlerts(testCase:XCTestCase) -> NSObjectProtocol {
+    return SnapshotAlertManager.automaticallyDismissAlerts(testCase)
+}
+
+func stopDismissingAlerts(testCase:XCTestCase) {
+    SnapshotAlertManager.stopDismissingAlerts(testCase)
+}
+
+func waitForAlertsToBeDismissed() {
+    SnapshotAlertManager.waitForAlertsToBeDismissed()
+}
+
+class SnapshotAlertManager: NSObject {
+    static var alertInterruptionMonitor:NSObjectProtocol?
+
+    class func stopDismissingAlerts(testCase:XCTestCase) {
+        if let previousDismissHandler = alertInterruptionMonitor {
+            testCase.removeUIInterruptionMonitor(previousDismissHandler)
+        }
+    }
+
+    class func automaticallyDismissAlerts(testCase:XCTestCase) -> NSObjectProtocol
+    {
+        SnapshotAlertManager.stopDismissingAlerts(testCase)
+        alertInterruptionMonitor = testCase.addUIInterruptionMonitorWithDescription("Snapshot alert auto hide") { (alert: XCUIElement) -> Bool in
```

Comment:
I've discovered that the UIInterruptionMonitor event is not triggered when the alert is showed but when the alert is blocking an action. So if an alert is displayed and no action is done, the alert will stay here forever. Hence my `XCUIApplication().tap()` after 2 seconds of waiting to trigger the method callback that will return false and close the alert.